pkg/sqlite/load.go

```diff
@@ -921,36 +928,61 @@ func (s *sqlLoader) addBundleProperties(tx *sql.Tx, bundle *registry.Bundle) err
 		if err != nil {
 			return err
 		}
-		if err := s.addProperty(tx, registry.GVKType, string(value), bundle.Name, bundleVersion, bundle.BundleImage); err != nil {
-			return err
+		properties[propstring{Type: registry.GVKType, Value: string(value)}] = struct{}{}
+	}
+
+	// Add properties from annotations
+	csv, err := bundle.ClusterServiceVersion()
+	if err != nil {
+		// FIXME: Returning nil here is in line with the original implementation, but that was probably wrong. We should probably just bubble-up the error.
+		return nil
+	}
+
+	if csv == nil {
+		// FIXME: Currently, a CSV is requirement of bundle addition. Should this return an error?
+		return nil
+	}
+
+	var props []registry.Property
+	if csv.GetAnnotations() != nil {
+		v, ok := csv.GetAnnotations()[registry.PropertyKey]
+		if ok {
+			if err := json.Unmarshal([]byte(v), &props); err != nil {
+				return err
+			}
 		}
```

Comment:
Not blocking, but I think I'd like to see this reversed and instead check if `!ok`:

```go
if !ok {
    continue
}

if err := json.Unmarshal(...);
...
```

Reply:
@timflannagan looks like this isn't in a `for` body, so `continue` won't do anything.

Reply:
Hmm, I was looking at a bad diff - I did a hard reload in my browser, and I see a different diff now 😵‍💫